assets/src/edit-story/elements/shared/useAverageColor.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import ColorThief from 'colorthief';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useCallback, useLayoutEffect } from '@wordpress/element';
+
+function useAverageColor( ref, onAverageColor ) {
+	const thief = useRef( new ColorThief() );
```

Comment:
This will still create an instance of the `ColorThief` each time. Are we certain it's cheap? Maybe `useMemo` is better here?

Reply:
I really don't know what I was thinking here. I've `const`'ed it outside the function now. I think it's fairly cheap.